plan/physical_plans.go

```diff
@@ -145,6 +145,9 @@ type PhysicalIndexScan struct {
 	accessEqualCount int
 
 	TableAsName *model.CIStr
+
+	// oldSchema is the original schema of DataSource.
```

Comment:
So why not name it `dataSourceSchema`, it would be more clear.
Please explain in comment what's different between the data source schema and the PhysicalIndexScan schema.